### ISSUE TITLE
CPU usage is too high

### ISSUE THREAD

**Issue by wind-hx** [gram-js/gramjs#351]
When I create 20 clients to forward messages at the same time, 1 core cpu reaches 100%
I don't know if it's a problem with the gramjs code or if nodejs can't make full use of the multi-core cpu
But I think the high cpu usage should be optimized

**Comment by DarkTwistter** [comment]
Any example on how to reproduce that?

**Comment by wind-hx** [comment]
> Any example on how to reproduce that?

I don't think this needs an example, 20 clients are created at the same time, and each client forwards 1000 messages, cpu rises immediately

**Comment by wind-hx** [comment]
https://tmpfiles.org/345145/cpu-profile-6942-20220420-174229.cpuprofile

I can provide cpuprofile

**Comment by DarkTwistter** [comment]
Imagine you are sending 20k post requests each second, thats not the easy task for cpu & connection interface

Have you also tried clustering to spread the load over cpu cores?

**Comment by wind-hx** [comment]
> Imagine you are sending 20k post requests each second, thats not the easy task for cpu & connection interface
> 
> Have you also tried clustering to spread the load over cpu cores?

I think it is 20 tcp connections continuously sending data

**Comment by wind-hx** [comment]
20 tcp connections continue to send messages will not cause the cpu to increase

**Comment by wind-hx** [comment]
I think it is necessary to optimize the encoding and receiving of a large amount of telegram transmission data

**Comment by bigbigsir** [comment]
you can use invokeWithoutUpdates

**Comment by bigbigsir** [comment]
Do you solve this problem?

**Comment by moeinrahimi1** [comment]
i have this issue of high cpu usage with only one client without even sending any message. running the script simply fills my cpu

**Comment by wind-hx** [comment]
> invokeWithoutUpdates

How to use invokeWithoutUpdates? Gramjs uses invokeWithLayer during initialization.

**Comment by johnhom1024** [comment]
I had this problem too, last logger show ：[INFO] - [Handling reconnect!], and cpu rising to 100% until stop. Network is good, It bother me for a long time, how to fix it.
